### DISCUSSION TITLE
[3차 스프린트 - 백엔드] 로깅 프레임 워크 및 모니터링 대시보드 조사

### DISCUSSION THREAD

**Discussion by Minjoo522** [woowacourse-teams/2024-devel-up#193] [백로그 논의]
#### 논의가 필요한 내용

로깅 프레임워크 및 (로그, 매트릭) 모니터링 대시보드 관련 조사
`8월 5일(월)` 공유 예정입니다.

조사 내용이나 정리 링크를 코멘트로 달아주세요.

**Comment by Minjoo522** [comment]
- [🚀 로깅 프레임워크](https://minjoo522.notion.site/070d25149c77401a85c6fa81b87af1d1?pvs=4)
- [🚀 모니터링 대시보드](https://minjoo522.notion.site/cd5c34110de84109b5bb8565616f0c2d?pvs=4)

**Comment by lilychoibb** [comment]
https://suave-bearskin-3ab.notion.site/19de9df19421458480dd73c97b46b92c?pvs=4

**Comment by le2sky** [comment]
[학습 내용 정리](https://sequoia-reading-664.notion.site/0eb31e9ec1514aceb56240b711b7be92)

**Comment by robinjoon** [comment]
[학습 내용](https://robinjoon.notion.site/8fb2635b453e4ad4a953004d89a5c5a9?pvs=4)

### 요약 

모니터링 : [spring-boot-actuator](https://github.com/spring-projects/spring-boot/tree/v3.2.7/spring-boot-project/spring-boot-actuator) + 프로메테우스 + 그라파나

로깅 : 스프링 부트 기본 의존성 그대로 사용. 예외, 요청 응답 HTTP 메세지 구성 요소, SQL 로그 필요하다고 생각.

**Comment by alstn113** [comment]
[로그](https://www.notion.so/alstn113/2024-08-04-4e7f18b098124b62807d4a800dd497c4?pvs=4): log4j가 logback보다 성능은 좋은 것 같은데 logback이 자료가 많은 것 같음. 

[모니터링](https://alstn113.notion.site/2024-08-04-73767eadfdae469c8465b6876c08f9c4): 액추에이터 + 프로메테우스 + 그라파나 + (옵션: 로키, 프롬테일), EC2 + Docker로 구동

**Comment by robinjoon** [comment]
> [학습 내용](https://robinjoon.notion.site/8fb2635b453e4ad4a953004d89a5c5a9?pvs=4)
> 
> ### 요약
> 모니터링 : [spring-boot-actuator](https://github.com/spring-projects/spring-boot/tree/v3.2.7/spring-boot-project/spring-boot-actuator) + 프로메테우스 + 그라파나
> 
> 로깅 : 스프링 부트 기본 의존성 그대로 사용. 예외, 요청 응답 HTTP 메세지 구성 요소, SQL 로그 필요하다고 생각.

추가로 메트릭 정보는 보안상 우리만 볼 수 있도록 조치를 취해야 한다!
